src/collect-all/Utilities.cs

```diff
@@ -0,0 +1,467 @@
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+//
+// Module: Utilities.cs
+//
+// Notes:
+// 
+// Utility class to help with setup and give common global methods. Note, this
+// class is a singleton.
+//
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using Newtonsoft.Json;
+using System.Runtime.InteropServices;
+
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+public class GeneralException : Exception
+{
+   public GeneralException() { }
+   
+   public GeneralException(string message) : base(message) { }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+public class Utilities
+{
+   /////////////////////////////////////////////////////////////////////////////
+   // Constructor(s)
+   /////////////////////////////////////////////////////////////////////////////
+   private Utilities(string tempPath = null, bool crossGen = false, bool loggingVerbose = false, string arch = null, string buildType = null, string coreclrRepoLocation = null) 
+   {  
+      // Find out what Operating system we are on. This is mostly just to
+      // make sure the OS is supported for SPMI Collection
+      string os = this.GetOs();;
+      
+      this.m_tempPath = tempPath;
+      this.m_crossGen = crossGen;
+      this.m_loggingVerbose = loggingVerbose;
+      
+      if (string.IsNullOrEmpty(arch) && string.IsNullOrEmpty(buildType) && string.IsNullOrEmpty(coreclrRepoLocation))
+      {
+         ParseSettingFile();
+
+         buildType = this.m_buildType;
+         coreclrRepoLocation = this.m_coreclrRepoLocation;
+         arch = this.m_arch;
+      }
+     
+      coreclrRepoLocation = this.m_coreclrRepoLocation;
+
+      // Set Default Values.
+      //
+      // jit: ryujit
+      // arch: x64
+      // buildType: Debug
+
+      if (string.IsNullOrEmpty(buildType))
+      {
+        buildType = "Debug";
+      }
+
+      if (string.IsNullOrEmpty(arch)) 
+      {
+         arch = RuntimeInformation.OSArchitecture.ToString();
```

Comment:
Why not just make this RuntimeInformation.OSArchitecture.ToString().ToLower() and then check against arch != "x64" to throw the exception?